File: Undergrad Work/Spring2014/CS243/Prog3/src/Stack.java
/**
This class represents a stack that contains the operands of the
expression. When the expression is completely processed, the result
is what is left in the stack.
@author Brianna Muleski
*/
public class Stack
{
   private Object[] elements;
   private int top;
   
   /**
   Constructor that initializes the stack and sets the top to zero.
   @param size the size of the stack
   */
   public Stack( int size )
   {
      elements = new Object[size];
      top = 0;
   }
   
   /**
   Checks if the stack is empty or not.
   @return true if empty, false otherwise
   */
   public boolean isEmpty()
   {
      return top == 0;
   }
   
   /**
   Checks if the stack is full or not.
   @return true if full, false otherwise
   */
   public boolean isFull()
   {
      return top == elements.length;
   }
   
   /**
   Adds an Object to the stack and increases top.
   @param x the Object being added
   */
   public void push( Object x )
   {
      elements[top++] = x;
   }
   
   /**
   Removes the last Object added.
   @return the top Object of the stack
   */
   public Object pop()
   {
      return elements[--top];
   }
   
   /**
   Clears the stack and resets top to zero.
   */
   public void clear()
   {
      for(int i = 0; i < elements.length; i++)
         elements[i] = null;
      top = 0;
   }
   
   /**
   Testbed main. Tests all the constructors and methods in this class.
   @param args is not used
   */
   public static void main ( String args[] ) 
   {
      //Test for contructor
      Stack s = new Stack(5);
      System.out.println("Test for constructor:");
      System.out.println("Expected: ");
      System.out.println("null null null null null");
      System.out.println("Result: ");
      for (int i = 0; i < s.elements.length; i++)
         System.out.print(s.elements[i] + " ");
      System.out.println();
      System.out.println();
      
      //Test for isEmpty method
      System.out.println("Test for isEmpty method:");
      if (s.isEmpty())
         System.out.println("isEmpty test passed.");
      else
         System.out.println("isEmpty test failed.");
      System.out.println();
      
      //Test for isFull method
      System.out.println("Test for isFull method:");
      if (s.isFull())
         System.out.println("isFull test failed.");
      else
         System.out.println("isFull test passed.");
      System.out.println();
      
      //Test for push method
      s.push(new Fraction("3/4"));
      s.push(new Fraction("1/2"));
      System.out.println("Test for push method:");
      System.out.println("Expected:");
      System.out.println("(0/1) (1/2)");
      System.out.println("Result:");
      for( int i = 0; i < s.top; i++ )
         System.out.print(s.elements[i] + " ");
      System.out.println();
      System.out.println();
       
      //Test for pop method
      System.out.println("Test for pop method:");
      System.out.println("Expected:");
      System.out.println("(1/2)");
      System.out.println("Result:");
      System.out.println(s.pop());
      System.out.println();
      
      //Test for clear method
      s.push(new Fraction("3/4"));
      s.push(new Fraction("5/8"));
      s.clear();
      System.out.println("Test for clear method:");
      System.out.println("Expected:");
      System.out.println("null null null null null");
      System.out.println("top = 0");
      System.out.println("Result:");
      for( int i = 0; i < s.elements.length; i++ )
         System.out.print(s.elements[i] + " ");
      System.out.println();
      System.out.println("top = " + s.top);
      System.out.println();
      
      
   }
}

File: Undergrad Work/Fall2014/OOPSII/Prog1/Stack.cpp
//-------------------------------------------------------------------
// Name:    Brianna Muleski   
// Project: 1 - Getting Started, Stack & Queue review
// Purpose: This class represents a stack. This class can initalize
//          the stack, clear the stack, check if the stack is empty
//          or full, push an item onto the top of the stack, and push
//          the top item off the stack.
//-------------------------------------------------------------------
#include "Stack.h"

SInfoType Stack::Pop()
{
   SInfoType item = items[--top];
   return item;
};

#ifdef TESTING_STACK

// ------------------------------------------------------------------
// Testbed main. Tests all the methods of the Stack class.
// ------------------------------------------------------------------
void main()
{
   Stack s;
   //Test for IsEmpty method with empty stack
   cout << "Tests for IsEmpty method:" << endl;
   if (s.IsEmpty())
      cout << "IsEmpty test passed" << endl;
   else
      cout << "IsEmpty test failed" << endl;

   //Test for IsEmpty method with not empty stack
   for (int i = 0; i < 5; i++)
      s.Push(2);
   if (s.IsEmpty())
      cout << "IsEmpty test failed" << endl << endl;
   else
      cout << "IsEmpty test passed" << endl << endl;

   //Test for IsFull method with empty stack
   s.Clear();
   cout << "Tests for IsFull method:" << endl;
   if (s.IsFull())
      cout << "IsFull test failed." << endl;
   else
      cout << "IsFull test passed." << endl;

   //Test for IsFull method with full stack
   for (int i = 0; i < MAXSTACK; i++)
      s.Push(4);
   if (s.IsFull())
      cout << "IsFull test passed." << endl << endl;
   else
      cout << "IsFull test failed." << endl << endl;

   //Test for Push method
   s.Clear();
   s.Push(4);
   cout << "Test for Push method:" << endl
        << "Expected:" << endl
        << "4" << endl
        << "Results:" << endl
        << s.Pop() << endl << endl;

   //Test for Pop method
   s.Push(4);
   s.Push(6);
   cout << "Test for Pop method:" << endl
        << "Expected:" << endl
        << "6" << endl
        << "Results:" << endl
        << s.Pop() << endl << endl;

   //Test for Clear method
   for (int i = 0; i < 10; i++)
      s.Push(i);
   s.Clear();
   cout << "Test for Clear method:" << endl;
   if (s.IsEmpty())
      cout << "Clear test passed" << endl;
   else
      cout << "Clear test failed" << endl;
};

#endif